Controllers/StaffController.cs

```diff
@@ -12,34 +13,29 @@ namespace stunning_robot_HR.Controllers
 {
     public class StaffController : Controller
     {
-        private readonly stunning_robot_HRContext _context;
-
-        public StaffController(stunning_robot_HRContext context)
+        private  IStaffRepository staffRepository;
+        public StaffController()
         {
-            _context = context;
+            this.staffRepository = new StaffRepository(new stunning_robot_HRContext());
         }
-
+        public StaffController(IStaffRepository staffRepository)
+        {
+            this.staffRepository = staffRepository;
+        }
+        
+        
         // GET: Staff
         public async Task<IActionResult> Index()
         {
-            return View(await _context.Staff.ToListAsync());
+             var staffs = from s in staffRepository.GetStaff()
```

Comment:
Just a little overly complex here. Something like var staff = staffRepository.GetStaff(); will do the trick.